backend/copr_backend/pulp.py

```diff
@@ -53,7 +52,20 @@ def url(self, endpoint):
         """
         A fully qualified URL for a given API endpoint
         """
-        return self.config["base_url"] + self.config["api_root"] + endpoint
+        domain = self.config["domain"]
+        if domain == "default":
+            domain = ""
+
+        relative = os.path.normpath("/".join([
+            self.config["api_root"],
+            domain,
+            endpoint,
+        ]))
```

Comment:
Since we use `os.path` for urls already (sub-optimal), don't you want to use `os.path.join()`?

Reply:
The problem with `os.path.join` is that it does something like this:

```python
>>> os.path.join("/foo/", "/bar/", "/baz/")
'/baz/'
```

while the `"/".join` variant produces something more expected

```python
ipdb> os.path.normpath("/".join(["/foo/", "/bar/", "/baz/"]))
'/foo/bar/baz'
```

Reply:
ah, OK